starlite/signature/parsing.py

```diff
@@ -108,7 +110,7 @@ def get_type_annotation_from_plugin(
 
 
 def parse_fn_signature(
-    fn: "AnyCallable", plugins: List["PluginProtocol"], dependency_name_set: Set[str]
+    fn: Union["AnyCallable", Type], plugins: List["PluginProtocol"], dependency_name_set: Set[str]
```

Comment:
I'm not sure this is necessary. Classes are already covered under `AnyCallable` - `callable` considers classes callables as well